docs/docs/developers/development/dialogs/index.md

```diff
@@ -64,4 +70,197 @@ It will result with following in UI dialog:
 
 ![Dialog example tab2](dialog-example-tab2.png)
 
-Explore available fields and options.
\ No newline at end of file
+## Data structure
+The data structure depends on the `name` property. 
+
+- In the simplest example property value is saved directly as a resource property. 
+```json
+"title": {
+  "sling:resourceType": "wcm/dialogs/components/textfield",
+  "label": "Title",
+  "name": "title"
+}
+```
+In above example, the `title` is added directly as a resource property.
+![Simple data property](dialog-data-simple.png)
+
+- There is also a possibility to manipulate data from other resources by using a path in the `name` property:
+```json
+"alt": {
+  "sling:resourceType": "wcm/dialogs/components/textfield",
+  "name": "image/alt",
+  "label": "Alt text"
+}
+```
+In above example, the `alt` property is saved as an image property from the header context.
+![Child resource property](dialog-data-path.png)
+
+- Another example of manipulating data from other resources is using a multifield. This dialog field `name` property defines a node under which are created other resources with properties defined in the multifield.
+```json
+"urls": {
+  "sling:resourceType": "wcm/dialogs/components/multifield",
+  "name": "urls",
+  "label": "Footer URLs",
+  "labelField": {
+    "sling:resourceType": "wcm/dialogs/components/textfield",
+    "name": "label",
+    "label": "Label"
+  }
+}
+```
+![Multifield resource property](dialog-data-multifield.png)
+
+## Validation
+Validation in dialog fields is used to verify if the value meets the criteria of the particular field. There are two levels:
+
+### FrontEnd validation
+Front End validation is done by Atlaskit by default, if the submitted value is obviously incorrect, like a non-number value in NumberField. You can find more details [here](https://atlassian.design/components).
```

Comment:
We are not using FE validation to avoid displaying validation messages in different way.